File: library/src/test/java/me/proxer/library/api/chat/UserChatRoomsEndpointTest.java
package me.proxer.library.api.chat;

import me.proxer.library.ProxerTest;
import me.proxer.library.api.ProxerException;
import me.proxer.library.entity.chat.ChatRoom;
import okhttp3.mockwebserver.MockResponse;
import org.junit.Test;

import java.io.IOException;
import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * @author Ruben Gees
 */
public class UserChatRoomsEndpointTest extends ProxerTest {

    @Test
    public void testDefault() throws IOException, ProxerException {
        server.enqueue(new MockResponse().setBody(fromResource("chat_rooms.json")));

        final List<ChatRoom> result = api.chat()
                .userRooms()
                .build()
                .execute();

        assertThat(result).first().isEqualTo(buildFirstTestRoom());
        assertThat(result).last().isEqualTo(buildLastTestRoom());
    }

    @Test
    public void testPath() throws ProxerException, IOException, InterruptedException {
        server.enqueue(new MockResponse().setBody(fromResource("chat_rooms.json")));

        api.chat().userRooms()
                .build()
                .execute();

        assertThat(server.takeRequest().getPath()).isEqualTo("/api/v1/chat/myrooms");
    }

    private ChatRoom buildFirstTestRoom() {
        return new ChatRoom("1", "Proxer.Me Hauptchat", "Willkommen im Proxer-Chat!",
                false, false);
    }

    private ChatRoom buildLastTestRoom() {
        return new ChatRoom("6", "Chat-Ankündigungen (21.03.18)", "",
                true, false);
    }
}

File: library/src/test/kotlin/me/proxer/library/api/chat/UserChatRoomsEndpointTest.kt
package me.proxer.library.api.chat

import me.proxer.library.ProxerTest
import me.proxer.library.entity.chat.ChatRoom
import me.proxer.library.runRequest
import org.amshove.kluent.shouldBeEqualTo
import org.junit.jupiter.api.Test

/**
 * @author Ruben Gees
 */
class UserChatRoomsEndpointTest : ProxerTest() {

    private val firstExpectedChatRoom = ChatRoom(
        id = "1",
        name = "Proxer.Me Hauptchat",
        topic = "Willkommen im Proxer-Chat!",
        isReadOnly = false
    )

    private val lastExpectedChatRoom = ChatRoom(
        id = "6",
        name = "Chat-Ankündigungen (21.03.18)",
        topic = "",
        isReadOnly = true
    )

    @Test
    fun testDefault() {
        val (result, _) = server.runRequest("chat_rooms.json") {
            api.chat
                .userRooms()
                .build()
                .safeExecute()
        }

        result.first() shouldBeEqualTo firstExpectedChatRoom
        result.last() shouldBeEqualTo lastExpectedChatRoom
    }

    @Test
    fun testPath() {
        val (_, request) = server.runRequest("chat_rooms.json") {
            api.chat.userRooms()
                .build()
                .execute()
        }

        request.path shouldBeEqualTo "/api/v1/chat/myrooms"
    }
}
